### DISCUSSION TITLE
Lua 5.3 for TurrisOS

### DISCUSSION THREAD

**Discussion by Fail-Safe** [sqm-autorate/sqm-autorate#85] [General] [answered]
Can any Turris users confirm if Lua 5.3 is available for install?

**Comment by Lochnair** [comment, marked as answer]
Doesn't look like it:
https://openwrt.org/packages/table_owrt19_7/start?dataflt[%25pageid%25*~]=lua5.3

And no other Lua packages except luajit in Turris' own repo: https://github.com/CZ-NIC/turris-os-packages/tree/master/lang

**Reply by Fail-Safe** [reply]
Thanks for confirming that @Lochnair! Too bad, but it is what it is.

**Reply by Lochnair** [reply]
Yeah, sadly Lua 5.3 would probably be difficult on OpenWrt 21.02 as well.
Looks like we'd need to modify the lualanes package to target Lua 5.3 as well to get that to work..

**Reply by trippleflux** [reply]
Also luarocks package for Lua 5.3?

**Reply by Lochnair** [reply]
Wow, I saw the issue you posted at the luarocks repo. I hadn't noticed the luarocks package in OpenWrt is that old (even in master). Luarocks 2.2.2 is from like April 2015.

I can understand them not wanting to debug such an old version.

We'll likely try to use LuaJIT for everyone if anything @trippleflux, that seems less troublesome right now, and it can be good for performance
